pkg/compute/rust.go

```diff
@@ -383,6 +384,48 @@ func (r Rust) Build(out io.Writer, verbose bool) error {
 	return nil
 }
 
+func (r *Rust) toolchainVersion(rustConstraint *semver.Constraints) error {
+	cmd := exec.Command("rustup", "toolchain", "list")
+	stdoutStderr, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("error executing rustup: %w", err)
+	}
+
+	scanner := bufio.NewScanner(strings.NewReader(string(stdoutStderr)))
+	scanner.Split(bufio.ScanLines)
+	var found bool
+	for scanner.Scan() {
+		// Example output for `rustup toolchain list`:
+		//
+		// stable-x86_64-apple-darwin (default)
+		// 1.46.0-x86_64-apple-darwin
+		// 1.49.0-x86_64-apple-darwin
+		v := strings.Split(scanner.Text(), "-")
+		if v[0] == "stable" {
+			continue
+		}
+
+		r.toolchain, err = semver.NewVersion(v[0])
+		if err != nil {
+			return fmt.Errorf("error parsing rust toolchain version: %w", err)
+		}
+
+		if ok := rustConstraint.Check(r.toolchain); ok {
```

Comment:
@kailan do you think we should go in _reverse_ for this logic? e.g. if my local rustup revealed...

```
stable-x86_64-apple-darwin (default)
1.49.0-x86_64-apple-darwin
1.50.0-x86_64-apple-darwin
```

...then it would mean that `1.49.0` would be used even though I have a newer rust version available (1.50.0) that's still within the supported range.

It's not massively important. It comes down to whether we feel it's worth pushing the user to the newest version of rust they happen to have available. So maybe something I revisit in a separate PR.

The downside though, is it would require more code to be written to support that selection process.

Thoughts?

Reply:
I believe we should always use the latest version available. I could see confusion arise when a user installs a new Rust version with rustup, but then doesn't actually see their applications being compiled with it – especially because this selection behaviour isn't visible to the user.

Alternatively, make it clear to the user on each `compute build` which version of Rust is being used.

Reply:
OR actually we could just let rustup decide which version to use and drop the `+1.49.0` flag from cargo commands. Then the behaviour can be configured using the `rustup default` command etc.